### ISSUE TITLE
TimeSynchronizedBatchSampler won't divide to batches

### ISSUE THREAD

**Issue by OS199301** [jdb78/pytorch-forecasting#246]
Hello,

I have some issues with the TimeSynchronizedBatchSampler.
When I'm generating the DataLoader using the to_dataloader function with batch_sampler="synchronized", and later on iterating over the batches I received, the target is always a scalar (the prediction length is 1 in my case), and the amount of batches equals the amount of samples (minus the look back / encoder length). 
In other words, each batch contains one sample.

When I use the to_dataloader without specifying an input for the batch_sampler, I do get batches, with the required amount of samples in the batch, and the sampler used is torch.utils.data.sampler.RandomSampler (the default sampler to my understanding).

Any idea on why is it now working and how to use the TimeSynchronizedBatchSampler correctly?
Alternatively, how can I use the torch SequentialSampler, which seems to be quite similar in purpose, in the to_dataloader function?

Thanks!!

**Comment by jdb78** [comment]
The time-synchronized sampler ensures that all samples are synchronized in time, i.e. have the same time index. If you have a unique time index, you will only have one sample per batch. If you want no shuffling, just use `shuffle=False`. What exactly is your use case?

**Comment by OS199301** [comment]
I'm trying to predict one time series based on itself and a few others, all share the same time index.
I want to iterate over the dataset, divided into batches, where the samples should remain causal. Ideally a stride enabling to make to data a little lighter would be great too.
As for the batches, I think that setting shuffle=false actually **solved my problem**, as now the BatchSampler used is the SequentialSampler, and the data is divided into batches the way i wanted it to be.

I actually have another question if you don't mind. 
As I said I have a single target time series, and a few time series as inputs, all share the same time.
I haven't figured out how to set the input correctly, in particular the group_id argument.
This is an example of my data set, where I'm trying to predict Close using Close and Open.
type is not needed, but was an attempt of mine to overcome the struggle.
How should I assign the inputs in this case?

Thank you very much!


![DataExample](https://user-images.githubusercontent.com/76777739/103800646-5432f500-5055-11eb-9dab-6190e7d31114.JPG)

**Comment by OS199301** [comment]
> I'm trying to predict one time series based on itself and a few others, all share the same time index.
> I want to iterate over the dataset, divided into batches, where the samples should remain causal. Ideally a stride enabling to make to data a little lighter would be great too.
> As for the batches, I think that setting shuffle=false actually **solved my problem**, as now the BatchSampler used is the SequentialSampler, and the data is divided into batches the way i wanted it to be.
> 
> I actually have another question if you don't mind.
> As I said I have a single target time series, and a few time series as inputs, all share the same time.
> I haven't figured out how to set the input correctly, in particular the group_id argument.
> This is an example of my data set, where I'm trying to predict Close using Close and Open.
> type is not needed, but was an attempt of mine to overcome the struggle.
> How should I assign the inputs in this case?
> 
> Thank you very much!
> 
> ![DataExample](https://user-images.githubusercontent.com/76777739/103800646-5432f500-5055-11eb-9dab-6190e7d31114.JPG)

Regarding my seconds question - I actually got the inputs right, apparently what confused me was that when looking at the data of a single batch, I could only recognize the target series as it was unnormalized, while all other timeseries were normalized.
